ops/pebble.py

```diff
@@ -2029,8 +2030,10 @@ def _wait_change_using_wait(self, change_id: ChangeID, timeout: Optional[float])
 
             try:
                 return self._wait_change(change_id, this_timeout)
-            except TimeoutError:
-                # Catch timeout from wait endpoint and loop to check deadline
+            except (socket.timeout, builtins.TimeoutError):
```

Comment:
Should we also catch `http.client.RemoteDisconnected` here?
See https://github.com/canonical/cos-lite-bundle/issues/103

Reply:
Based on the stack trace I think it's necessary. Thoughts? @benhoyt @tonyandrewmeyer

Reply:
Without further investigation, I don't think we should add this (or other exception types here). We're specifically catching the various kinds of timeout, but I don't think we want to start catching other errors in general. I'm not exactly sure what `RemoteDisconnected` is or is caused by, but it sounds like something the remote side did actively, not something we want to automatically retry for.